src/Symfony/Component/PropertyInfo/Tests/Extractors/ReflectionExtractorTest.php

```diff
@@ -0,0 +1,122 @@
+use Symfony\Component\PropertyInfo\Type;
+
+/**
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ReflectionExtractorTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var ReflectionExtractor
+     */
+    private $extractor;
+
+    public function setUp()
+    {
+        $this->extractor = new ReflectionExtractor();
+    }
+
+    public function testGetProperties()
+    {
+        $this->assertEquals(
+            array(
+                'bal',
+                'parent',
+                'collection',
+                'foo',
+                'foo2',
+                'foo3',
+                'foo4',
+                'foo5',
+                'files',
+                'A',
+                'B',
+                'C',
+                'D',
+                'E',
+                'F',
+            ),
+            $this->extractor->getProperties('Symfony\Component\PropertyInfo\Tests\Fixtures\Dummy')
+        );
+    }
+
+    /**
+     * @dataProvider typesProvider
+     */
+    public function testExtractors($property, array $type = null)
+    {
+        $this->assertEquals($type, $this->extractor->getTypes('Symfony\Component\PropertyInfo\Tests\Fixtures\Dummy', $property, array()));
+    }
+
+    public function typesProvider()
+    {
+        return array(
+            array('a', null),
+            array('b', array(new Type(Type::BUILTIN_TYPE_OBJECT, true, 'Symfony\Component\PropertyInfo\Tests\Fixtures\ParentDummy'))),
+            array('c', array(new Type(Type::BUILTIN_TYPE_BOOL))),
+            array('d', array(new Type(Type::BUILTIN_TYPE_BOOL))),
+            array('e', null),
+            array('f', array(new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_OBJECT, false, 'DateTime')))),
+        );
+    }
+
+    /**
+     * @dataProvider php7TypesProvider
+     */
+    public function testExtractPhp7Type($property, array $type = null)
+    {
+        if (!method_exists('\ReflectionMethod', 'getReturnType')) {
+            $this->markTestSkipped('Available only with PHP 7 and superior.');
+        }
+
+        $this->assertEquals($type, $this->extractor->getTypes('Symfony\Component\PropertyInfo\Tests\Fixtures\Php7Dummy', $property, array()));
+    }
+
+    public function php7TypesProvider()
+    {
+        return array(
+            array('foo', array(new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true))),
+            array('bar', array(new Type(Type::BUILTIN_TYPE_INT))),
+            array('baz', array(new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT), new Type(Type::BUILTIN_TYPE_STRING)))),
+        );
+    }
+
+    public function testIsReadable()
+    {
+        $this->assertFalse($this->extractor->isReadable('Symfony\Component\PropertyInfo\Tests\Fixtures\Dummy', 'bar', array()));
+        $this->assertFalse($this->extractor->isReadable('Symfony\Component\PropertyInfo\Tests\Fixtures\Dummy', 'baz', array()));
```

Comment:
i'm not sure, but i saw in some symfony test that we use something like "->foo() should assert false" as custom message